### PR TITLE
Misc doc updates

### PR DESCRIPTION
**Pull request by jayteemo**
Improve documentation in header comments.

### CODE REVIEW

**Comment by jayteemo** [comment]
The changes were requested by Tuomas: https://github.com/theflintstone/sdk-nrf/pull/1

**Review by plskeggs — APPROVED** [review]

File: include/net/nrf_cloud_coap.h [hunk 1]
```diff
@@ -61,6 +61,11 @@ int nrf_cloud_coap_init(void);
 
 /**
  * @brief Connect to and obtain authorization to access the nRF Cloud CoAP server.
+ * The full DTLS handshake will be performed, and on success, a connection ID (CID) will be
```
Comment (peknis) [inline include/net/nrf_cloud_coap.h:64]:
```suggestion
 * The full DTLS handshake is performed, and on success, a connection ID (CID) is
```

File: include/net/nrf_cloud.h [hunk 2]
```diff
@@ -632,10 +639,11 @@ struct nrf_cloud_ctrl_data {
...
 	bool alerts_enabled;
-	/** If 0, the nrf_cloud library logging backend will be disabled.
-	 *  If values from 1 to 4, this level and any lower levels will
-	 *  be sent to the cloud. Level 1 is most urgent (LOG_ERR),
-	 *  level 4 least (LOG_DBG).
+	/** If 0: None - the nrf_cloud library logging backend will be disabled.
+	 *     4: LOG_DBG (least urgent) and all levels below will be sent to the cloud.
+	 *     3: LOG_INF and all levels below will be sent to the cloud.
+	 *     2: LOG_WRN and all levels below will be sent to the cloud.
+	 *     1: only LOG_ERR (most urgent) will be sent to the cloud.
```
Comment (peknis) [inline include/net/nrf_cloud.h:646]:
```suggestion
	/** If 0: None - the nrf_cloud library logging backend is disabled.
	 *     4: LOG_DBG (least urgent) and all levels below are sent to the cloud.
	 *     3: LOG_INF and all levels below are sent to the cloud.
	 *     2: LOG_WRN and all levels below are sent to the cloud.
	 *     1: only LOG_ERR (most urgent) is sent to the cloud.
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: include/net/nrf_cloud_coap.h [hunk 3]
```diff
@@ -364,7 +372,8 @@ int nrf_cloud_coap_shadow_state_update(const char * const shadow_json);
 int nrf_cloud_coap_shadow_desired_update(const char * const shadow_json);
 
 /**
- * @brief Update the device status in the shadow's reported state section.
+ * @brief Update the device's reported shadow section with information about the device, network,
+ * modem, SIM card and FOTA capabilities.
```
Comment (peknis) [inline include/net/nrf_cloud_coap.h:376]:
```suggestion
 * modem, SIM card, and FOTA capabilities.
```

File: include/net/nrf_cloud.h [hunk 4]
```diff
@@ -451,7 +451,12 @@ struct nrf_cloud_svc_info_fota {
 	uint8_t _rsvd:3;
 };
 
-/** @brief DEPRECATED - No longer used by nRF Cloud */
+/** @brief DEPRECATED - No longer used by nRF Cloud
+ * The device data cards on nrfcloud.com automatically appear when the device
```
Comment (peknis) [inline include/net/nrf_cloud.h:455]:
```suggestion
 * The device data cards on the nRF Cloud portal automatically appear when the device
```

**Review by peknis — APPROVED** [review]
Approved with a couple of minor nits.

**Review by glarsennordic — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by eivindj-nordic — APPROVED** [review]

**Pull request merged**